File: week1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;

    do
    {
        change = get_float("Change owed: ");
    } while (change < 0);

    int cents = round(change * 100);
    int coins = 0;
    while (cents >= 25)
    {
        cents = cents - 25;
        coins++;
    }
    while (cents >= 10)
    {
        cents = cents - 10;
        coins++;
    }
    while (cents >= 5)
    {
        cents = cents - 5;
        coins++;
    }
    while (cents >= 1)
    {
        cents = cents - 1;
        coins++;
    }

    printf("%i\n", coins);
}

File: week6/pset6/cash/cash.py
from cs50 import get_float

# ask user until non-negative value is provided
dollars = 0
while dollars <= 0:
    dollars = get_float("Change owed: ")

# convert the dollars into cents
cents = int(round(dollars * 100))

# counts the no of coins for change
coincount = 0
while (cents > 0):
    if cents >= 25:
        cents -= 25
        coincount += 1
    elif cents >= 10:
        cents -= 10
        coincount += 1
    elif cents >= 5:
        cents -= 5
        coincount += 1
    elif cents >= 1:
        cents -= 1
        coincount += 1

print(f"{coincount}")